### PR TITLE
Starting to put together Github API test

### PR DESCRIPTION
**Pull request by DillonSimeone**
Worked my way though some more svelte tutorials tonight.

Changed the signin method to take a provider arg, so different processes can be put in place easily for each types of accounts.

Made notes in GithubTest.svelte for future me to chew on tomorrow night. (Or someone else if they're feeling EXTRA hungry!)

I ran npx standard and npx standard --fix... It look like some weird changes was made by that thingy. Probably all good.

### CODE REVIEW

File: src/pages/Setup.svelte [hunk 1]
```diff
@@ -1,41 +1,54 @@
...
-import '../firebase.js'
-import '@firebase/auth'
-import { Link } from 'svelte-routing'
-export const language = 'en'
+  import Layout from "../components/Layout.svelte";
+  import firebase from "@firebase/app";
+  import "../firebase.js";
+  import "@firebase/auth";
+  import { Link } from "svelte-routing";
+  export const language = "en";
```
Comment (Shiroraven) [inline src/pages/Setup.svelte]: These additions are the result of `standard --fix` not working on svelte files, please check all files before submitting a pull request. 

Currently there is no known way of automatically fixing those issues inside `.svelte` files I am sorry about that.

File: src/pages/Setup.svelte [hunk 2]
```diff
@@ -1,41 +1,54 @@
...
 
-const step = 1
-let user = ''
+  const step = 1;
+  let user = "";
```
Comment (Shiroraven) [inline src/pages/Setup.svelte]: here again, please use single quotes and no semicolon

File: src/pages/Setup.svelte [hunk 3]
```diff
@@ -1,41 +1,54 @@
...
-    .auth()
-    .signInWithPopup(provider)
-}
+    if (u) {
+      user = u;
+      provider = u.providerData[0].providerId;
+      console.log(u); // Debugging
+      console.log(
+        "Mmm.... I think this account is a " + provider + " account!"
+      ); // Debugging
```
Comment (Shiroraven) [inline src/pages/Setup.svelte]: when submitting test cases, or debug functions please comment them out before submission.

File: src/pages/Setup.svelte [hunk 4]
```diff
@@ -127,14 +141,14 @@ const signout = () => {
       <label
         for="lang_english"
         class={language == 'en' ? 'lang active' : 'lang'}
-        on:click={() => language = 'en'}>
+        on:click={() => (language = 'en')}>
```
Comment (Shiroraven) [inline src/pages/Setup.svelte]: I am not sure if your formatter is doing this automatically, but if that is the case lets avoid it.

File: src/pages/Setup.svelte [hunk 5]
```diff
@@ -151,16 +165,33 @@ const signout = () => {
...
       Signed in as
       <h1>{user.displayName}</h1>
       <button on:click={signout}>sign out</button>
-      <Link to="/test">Google API Test</Link>
+      {#if provider === 'google.com'}
+        <Link to="/testGapi">Google API Test</Link>
+      {/if}
+      {#if provider === 'github.com'}
+        <Link to="/testGithubApi">Github API Test</Link>
+      {/if}
```
Comment (Shiroraven) [inline src/pages/Setup.svelte]: I really like the way you did this. Didn't know svelte-if is stackable.

File: src/pages/Setup.svelte (hunk 4 above)
Comment (DillonSimeone) [reply src/pages/Setup.svelte]: Yep, that was what caught my eyes right away. Made me wonder if I was using standard --fix right... Weird change. I was surprised to find that those changes didn't break the website!

File: src/pages/Setup.svelte (hunk 3 above)
Comment (DillonSimeone) [reply src/pages/Setup.svelte]: All right, I'll do that from now and on. I mostly left that in to get you onto the same page of recognizing the importance of detecting differences between providers without having to say anything about it.

File: src/pages/Setup.svelte (hunk 1 above)
Comment (DillonSimeone) [reply src/pages/Setup.svelte]: Time to flog Standard --fix for this! Down to the bones! I'll manually format those for now then look into finding a way to force Standard --fix or something to play nice with Svelte later tonight.

File: src/pages/Setup.svelte (hunk 2 above)
Comment (DillonSimeone) [reply src/pages/Setup.svelte]: This one, Standard --fix can't be blamed for! All on my neck. Smelly habits.

File: src/pages/Setup.svelte (hunk 3 above)
Comment (DillonSimeone) [reply src/pages/Setup.svelte]: Idea, what do you think of setting up a debug flag?

```
const debug = true
...
if(debug){
console.log blah blah
}
```

**Pull request merged**